ouroboros-consensus-cardano/test/Test/ThreadNet/TxGen/Cardano.hs

```diff
@@ -1,15 +1,308 @@
+
+  type TxGenExtra (CardanoBlock c) = CardanoTxGenExtra c
+
+  -- TODO also generate " typical " Byron and Shelley transactions
+  testGenTxs (CoreNodeId i) _ncn curSlot cfg extra ls =
+      pure $ maybeToList $ migrateUTxO migrationInfo curSlot lcfg ls
+    where
+      lcfg = blockConfigLedger $ topLevelConfigBlock cfg
+
+      CardanoTxGenExtra
+        { ctgeByronGenesisKeys
+        , ctgeNetworkMagic
+        , ctgeShelleyCoreNodes
+        } = extra
+
+      GeneratedSecrets
+        { gsRichSecrets
+        } = ctgeByronGenesisKeys
+
+      migrationInfo = MigrationInfo
+        { byronMagic = ctgeNetworkMagic
+        , byronSK
+        , paymentSK
+        , poolSK
+        , stakingSK
+        , vrfSK
+        }
+
+      byronSK :: Byron.SigningKey
+      byronSK = gsRichSecrets !! fromIntegral i
+
+      Shelley.CoreNode
+        { Shelley.cnDelegateKey = paymentSK
+        , Shelley.cnStakingKey  = stakingSK
+        , Shelley.cnVRF         = vrfSK
+        } = ctgeShelleyCoreNodes !! fromIntegral i
+
+      -- Reuse the payment key as the pool key, since it's an individual
+      -- stake pool and the namespaces are separate.
+      poolSK :: SignKeyDSIGN (DSIGN c)
+      poolSK = paymentSK
+
+-- | See 'migrateUTxO'
+data MigrationInfo c = MigrationInfo
+  { byronMagic :: Byron.NetworkMagic
+    -- ^ Needed for creating a Byron address.
+  , byronSK    :: Byron.SigningKey
+    -- ^ The core node's Byron secret.
+  , paymentSK  :: SignKeyDSIGN (DSIGN c)
+  , poolSK     :: SignKeyDSIGN (DSIGN c)
+  , stakingSK  :: SignKeyDSIGN (DSIGN c)
+  , vrfSK      :: SignKeyVRF (VRF c)
+    -- ^ To be re-used by the individual pool.
+  }
+
+-- | Convert a core node's utxo from Byron to an active Shelley stake pool.
+--
+-- Returns a transaction that registers a staking key, registers an individual
+-- stake pool, delegates that stake key to that stake pool, and transfers all
+-- utxo from the Byron 'byronAddr' to the Shelley address corresponding to the
+-- pair of 'paymentSK' and 'stakingSK'.
+--
+-- It returns 'Nothing' if the core node does not have any utxo in its
+-- 'byronAddr' (eg if this transaction has already been applied).
+migrateUTxO :: forall c.
+               (TPraosCrypto c, DSIGN c ~ Ed25519DSIGN)
+            => MigrationInfo c
+            -> SlotNo
+            -> LedgerConfig (CardanoBlock c)
+            -> LedgerState (CardanoBlock c)
+            -> Maybe (GenTx (CardanoBlock c))
+migrateUTxO migrationInfo curSlot lcfg lst
+    | Just utxo <- mbUTxO =
+
+    let picked :: Map (SL.TxIn c) (SL.TxOut c)
+        picked =
+            Map.filter pick $ SL.unUTxO utxo
+          where
+            pick (SL.TxOut addr _) =
+                addr == SL.AddrBootstrap (SL.BootstrapAddress byronAddr)
+
+        -- Total held by 'byronAddr'
+        pickedCoin :: SL.Coin
+        pickedCoin =
+            sum $ fmap (\(SL.TxOut _ coin) -> coin) picked
+
+        fee        :: SL.Coin
+        fee        = 0
```

Comment:
I guess you just have `minFee` and `poolDeposit` set to 0?

Reply:
Yeah, the `Cardano` ThreadNet uses the `ouroboros-shelley-test` genesis config

https://github.com/input-output-hk/ouroboros-network/blob/83baf7eb58907137d50605fc2c85fd4c51ae68e0/ouroboros-consensus-cardano/test/Test/ThreadNet/Cardano.hs#L446-L447

which boils down to `SL.emptyPParams`.

https://github.com/input-output-hk/ouroboros-network/blob/62e57b9673e0140c87c6edd6ca8fb7b813c9f8d9/ouroboros-consensus-shelley-test/src/Test/ThreadNet/Infra/Shelley.hs#L293